data/capture_config.yaml

```diff
@@ -0,0 +1,52 @@
+spoof_regex: .*
+ntlm_challenge: "1122334455667788"
+ntlm_domain: anonymous
+http_basic: no
+ssl_cert: null
```

Comment:
If moved into per service consider all hardcoded defaults could move into config and the `configure_*` methods could be consolidated to iterate the proposed `datastore` map in a meta-programing format similar to:
```
def configure_module(datastore, config)
  config[:datastore].each do |option|
    datastore[option.to_s] = config[:datastore][option]
  end
end
```

Reply:
There's no particular reason they should be global; just an artifact of the earlier request to have the command line options defaults be pulled from a file. That said, even if we did it per-service, I think having an overall default makes sense, to provide a default in its absence.

I can see how this suggested approach would be much more flexible; but I'm weighing up flexibility vs usability in my mind. Given a fixed NTLM challenge is used for rainbow tables, would anyone want to provide a different value for each different service? Would there be an opsec use case for different NTLM domains for each service (I'd think someone would probably want the same for all of them)? Is providing a different TLS cert for each service a valuable thing? (I'd think that 99% of the time, we just want them to be the same).

To me, this feature is most useful because it packages everything up neatly for the common use case, and it's easy for someone to understand how to use it. I should clarify: I'm happy to make that change if you think it's important; just presenting my view as an operator.

If we did want to go ahead, though, the question to answer is: how do these settings interact with the individual command line options? Would providing a command line option for SSLCert override all of the individual datastore settings in the config file?

Reply:
This is sound reasoning to me. I think having the a default in place is good, customizing the services down the line can be another iteration.

The `ntlm_challenge` value seems like a `fingerprintable` item that may be worth thinking more on in the future. As things iterate I could see `http_basic` for a configured service being something a user might want to very and fallback to the global default for some services.

Shifting `hardcoded` port values and such currently in the `configure_*` methods may help users understand what is running at a glance in the config instead of having to view the plugin source. That also can be deferred as an enhancement.